Enterprise-CMCS/macpro-mdct-mfp
# Added Alert Check To Statusing

### Description

For the State & Territory Specific Initiatives, statusing was only partial completed due to the special addition of extra conditions indicated in the alert box. 

I tried capturing the alert state changes at first in the fieldData and then in the metadata, but after much trial and error, it was best to not store it as saved data. Instead, I have a function that checks to make sure there's no frontend blockers to submitting. This function does not do data validation, that is still left to the completeStatus function in the backend.

Note: Easiest way to check is, if the alert box is active in "State and Territory-Specific Initiatives", the status in Review & Submit should always be an error and red.


https://github.com/Enterprise-CMCS/macpro-mdct-mfp/assets/127151429/3be82cdc-205c-44a5-8adf-45c6f9fba8e3



### Related ticket(s)


CMDCT-400

---
### How to test

1) Sign into MFP as a state user
2) Create a new WP
3) Go to "State and Territory-Specific Initiatives", add and complete 1 Initiative
4) Go to Review & Submit
5) The status for State & Territory Specific Initiatives = Error as the conditions haven't been met even through 1 is filled out
6) Follow the conditions set by the alert box and fill out the initiatives
7) The status in Review & Submit should be a check mark

### Important updates


N/A

---
### Author checklist

---


_convert to a different template: [test → val](?expand=1&template=test-to-val-deployment.md)_ | _[val → prod](?expand=1&template=val-to-prod-deployment.md)_
